### PR TITLE
Create and use Git wrapper

### PR DESCRIPTION
**Pull request by attack**
- allows overriding of git functionality to be isolated in git wrapper and not throughout codebase

### CODE REVIEW

**Comment by attack** [comment]
@lpender Just want to push where am I headed with this and see if this idea of wrapping git fits your vision for the project.

File: lib/bummr/git.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  class Git
+    include Singleton
+    include Log
+
+    def initialize
+      @git_commit = ENV.fetch("BUMMR_GIT_COMMIT") { "git commit" }
+    end
+
+    def commit(message)
+      system("#{git_commit} -am '#{message}'")
```
Comment (lpender) [inline lib/bummr/git.rb]: Can we make it just `git commit -m"#{message}"` and drop the `a` ?

Probably better to do the adding and the committing separately if we're going to use this kind of pattern.

Comment (lpender) [reply lib/bummr/git.rb]: For example, you can see that in my latest PR, I only add `Gemfile` and `Gemfile.lock` (I noticed sometimes it was committing logs that were generated)

https://github.com/lpender/bummr/pull/51/files#diff-dd5832feaa364e94da7df94cd3c2f8f6R39

**Comment by attack** [comment]
@lpender updated with `git add` suggestions.

Based on your comments, it seems you are okay with where this is going so I will continue extracting git commands.

**Comment by attack** [comment]
I have also been experimenting with running `bummr update` in headless mode.
https://github.com/attack/bummr/pull/1

**Comment by lpender** [comment]
Hi @attack -- I'm afraid some of my upstream changes caused conflicts. If you want to take a look at resolving those conflicts I'll take another look at this.

I took a quick pass and it's looking pretty good.

Thanks!

**Comment by attack** [comment]
@lpender updated after merge conflicts

**Pull request merged**